llm/server.go

```diff
@@ -864,6 +865,52 @@ func (s *llmServer) Embedding(ctx context.Context, prompt string) ([]float64, er
 		return nil, fmt.Errorf("unexpected server status: %s", status.ToString())
 	}
 
+	switch prompts := prompts.(type) {
+	case string:
+		// single prompt
+		embedding, err := s.EmbeddingSingle(ctx, prompts)
+		if err != nil {
+			return nil, err
+		}
+		return [][]float64{embedding}, nil
+	case []string:
+		// multiple prompts
+		errCh := make(chan error, 1)
```

Comment:
See my previous PR where we handle the batch directly in server.cpp

While this moves some of the concurrency to Go it will be slower – instead we should build up the batch of all the prompts and decode them all at once in server.cpp instead of this loop